tests/unit/services/test_remotebuild.py

```diff
@@ -146,25 +146,47 @@ def test_is_project_private_error(remote_build_service):
         remote_build_service.is_project_private()
 
 
+@pytest.mark.parametrize(
+    ("information_type", "push_url"),
+    [
+        # exercise one public type and one private type
+        (
+            launchpad.models.InformationType.PUBLIC,
+            "https://craft_test_user:super_secret_token@git.launchpad.net/~user/+git/my_repo",
+        ),
+        (
+            launchpad.models.InformationType.PROPRIETARY,
```

Comment:
So in the Launchpad UI, I can set a project to public or proprietary, but I can set a repo to public or private (or public/private security). I don't actually know what happens in the API if we try to create a repo with information type proprietary.

Reply:
Is it not possible via the API?

I'm looking at `information_type` in https://api.launchpad.net/devel.html#git_repository

Reply:
Here's my project's visibility settings and the repo and recipe that the remote builder creates for that project:

![image](https://github.com/user-attachments/assets/8624ab87-359d-4465-ac28-ef768ffd2667)


![image](https://github.com/user-attachments/assets/5faec9a4-7ee0-4466-b117-e0639f49d35f)

Reply:
Here's what I'm referring to:

1. Project: 
![image](https://github.com/user-attachments/assets/7e1c2ef1-1d3e-4fb8-bc31-d91291b0c591)

2. 
![image](https://github.com/user-attachments/assets/1ebc815d-0fe8-42cf-97bd-cb6450c01e1e)


This seems to still leak repository names. If I go to https://code.launchpad.net/~mr-cal/+git I can see:

![image](https://github.com/user-attachments/assets/16544e91-f53a-469b-b1de-40a446812436)

though if I try to access any of them under the private project it's denied. 

I don't know whether setting private instead of proprietary for the repository will hide altogether, but if it does we should prefer that.

Reply:
Note from call: this does work correctly - the old repos are from testing during development.